fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -239,11 +252,67 @@ def __init__(
     def module(self) -> nn.Module:
         return self._fsdp_wrapped_module  # note: may be a FlattenParamsWrapper instance
 
+    def apply(self, fn: Callable[[nn.Module], None]) -> "FullyShardedDataParallel":
+        """
+        Applies ``fn`` recursively to every submodule (as returned by
+        ``.children()``) as well as self. Typical use includes initializing the
+        parameters of a model.
+
+        Compared to ``torch.nn.Module.apply``, this version additionally gathers
```

Comment:
we should mention what training state should this be called in. I guess only idle state. It would be weird if this is called in the summon state, right?